src/remote/plc/testing.rs

```diff
@@ -0,0 +1,535 @@
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::iter;
+
+use atrium_api::types::string::{Cid, Datetime, Did};
+use atrium_crypto::keypair::{Did as _, Export, P256Keypair};
+use base64ct::Encoding;
+use rand_core::OsRng;
+
+use super::{AuditLog, ChangeOp, LegacyCreateOp, LogEntry, Operation, SignedOperation};
+use crate::{
+    data::{PlcData, Service},
+    util::derive_did,
+};
+
+/// The state of an identity as of a particular operation.
+struct Identity {
+    rotation: Vec<P256Keypair>,
+    signing: HashMap<String, P256Keypair>,
+}
+
+impl Identity {
+    fn generate() -> Self {
+        let mut rng = OsRng;
+
+        Self {
+            rotation: vec![P256Keypair::create(&mut rng), P256Keypair::create(&mut rng)],
+            signing: iter::once(("atproto".into(), P256Keypair::create(&mut rng))).collect(),
+        }
+    }
+}
+
+pub(crate) struct TestLog {
+    initial_state: Identity,
+    state_updates: Vec<(usize, Identity)>,
```

Comment:
TODO: Decide on a test kit API that enables `apply_update` to add / change / remove keys.